Doc Examples

philkra opened the pull request:
* final directory structure
* first stab at generator

nik9000 commented on examples/docs/generator/generate.php:
```diff
@@ -0,0 +1,199 @@
+
+    // Generate Asciifdoc Include Files
+    foreach($examples as $example) {
+        if(file_exists(sprintf('%s/%s.asciidoc', $asciidir, $example['hash'])) === false && file_exists(sprintf('%s/_%s.asciidoc', $asciidir, $example['hash'])) === false) {
```
So you make a `_` file which will resolve the include and just remove the `_` when you are ready. Clever.

nik9000 replied:
I'm starting to prefer the `.adoc` suffix over `.asciidoc`. Both are standard to be honest and you can totally do whichever.

philkra replied:
your choice, will rename the files to `.adoc`

philkra commented on examples/docs/generator/helpers.php:
```diff
@@ -0,0 +1,12 @@
+<?php
+
+// === getting-started.asciidoc: line 209: f8cc4b331a19ff4df8e4a490f906ee69
```
yeah, the use of the whole file is yet to be determined or to say it frankly, I will delete it :)

ezimuel commented on examples/docs/src/Examples/Docs/Aggregations/Bucket/AdjacencyMatrixAggregation.php:
```diff
@@ -12,7 +12,7 @@
  * Class: AdjacencyMatrixAggregation
  *
- * Date: 2019-08-05 08:49:19
+ * Date: 2019-08-06 06:56:02
```
Is the date important here? If we run the generator and the code is the same we'll have different files with just a date diff.  This is something expected or can we remove it to avoid superfluous commits?

nik9000 replied:
I do like the idea of keeping the diff small!

philkra replied:
no it's not, you are 100% correct on the diff, that bit me already a couple of times. will remove it

philkra commented:
Summary:
* add new line to asciidoc skeleton
* changed the file extension to `.adoc` in the generator and the existing files
* removed the timestamp from the classes and the methods
* deleted the incredibly useless `helpers.php` file :)
